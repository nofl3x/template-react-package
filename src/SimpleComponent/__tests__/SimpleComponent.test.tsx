import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import { ISimpleComponentProps, SimpleComponent } from '../index'

const defaultProps: ISimpleComponentProps = {
  text: 'Simple text'
}

describe('<SimpleComponent /> Snapshot', () => {

  describe('SimpleComponent Snapshot', () => {
    afterEach(cleanup)

    it('It should take a snapshot', () => {
      const { asFragment } = render(<SimpleComponent {...defaultProps} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })


  describe('SimpleComponent props', () => {
    afterEach(cleanup)

    it('It should render text correctly', () => {
      const { getByTestId } = render(<SimpleComponent {...defaultProps} />)
      expect(getByTestId('test-text')).toHaveTextContent(defaultProps.text)
    })
  })
})


