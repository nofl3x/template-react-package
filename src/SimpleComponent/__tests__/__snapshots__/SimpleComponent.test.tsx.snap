// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleComponent /> Snapshot SimpleComponent Snapshot It should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="simpleComponent"
  >
    <div
      class="simpleComponent__text"
      data-testid="test-text"
    >
      Simple text
    </div>
    <div
      class="simpleComponent__content"
    />
  </div>
</DocumentFragment>
`;
